src/smart-components/role/add-role-permissions/add-role-permission-wizard.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import AddRolePermissionView from './add-role-permission-view';
+import AddRolePermissionSummaryContent from './add-role-permissions-summary-content';
+import PropTypes from 'prop-types';
+import { WarningModal } from '../../common/warningModal';
+import { useHistory } from 'react-router-dom';
+import { Wizard } from '@patternfly/react-core';
+import { updateRole } from '../../../helpers/role/role-helper';
+
+const AddRolePermissionWizard = ({ role }) => {
+  console.log('Testing out my role as soon as the Wizard loads: ', role);
+  const [selectedPermissions, setSelectedPermissions] = useState([]);
+  const [cancelWarningVisible, setCancelWarningVisible] = useState(false);
+  const [currentRole, setCurrentRole] = useState({});
+  const [currentRoleID, setCurrentRoleID] = useState('');
+  const history = useHistory();
+
+  const setSelectedRolePermissions = (selected) => {
+    setSelectedPermissions(selected);
+  };
+
+  useEffect(() => {
+    setCurrentRole(role);
+    setCurrentRoleID(role.uuid);
+  }, [selectedPermissions]);
+
+  const steps = [
+    {
+      id: 1,
+      name: 'Add Permissions',
+      component: new AddRolePermissionView({ selectedPermissions, setSelectedRolePermissions, role }),
+    },
+    {
+      id: 2,
+      name: 'Review Details',
+      component: new AddRolePermissionSummaryContent({ selectedPermissions, role }),
+      nextButtonText: 'Confirm',
+    },
+  ];
+
+  const handleWizardCancel = () => {
+    setCancelWarningVisible(true);
+  };
+
+  const handleConfirmCancel = () => {
+    history.goBack();
+  };
+
+  const onSubmit = async () => {
+    const cleanPermissions = selectedPermissions.map((permission) => {
+      // don't have info on how we populate resourceDefinitions
+      return { resourceDefinitions: [], permission: permission.uuid };
+    });
```

Comment:
No need to wrap in function body
```suggestion
    const cleanPermissions = selectedPermissions.map(({ uuid } = {}) => ({
      resourceDefinitions: [], permission: uuid
    });
```

Reply:
@arivepr^